src/main/java/com/autotune/analyzer/recommendations/model/PerformanceBasedRecommendationModel.java

```diff
@@ -436,11 +440,32 @@ public Map<AnalyzerConstants.RecommendationItem, RecommendationConfigItem> getAc
             return null;
         }
 
-        double coreAverage = CommonUtils.percentile(PERFORMANCE_ACCELERATOR_PERCENTILE, acceleratorCoreMaxValues);
-        double memoryAverage = CommonUtils.percentile(PERFORMANCE_ACCELERATOR_PERCENTILE, acceleratorMemoryMaxValues);
+        // Return null if entries are empty
+        if (acceleratorCoreMaxValues.isEmpty() && acceleratorMemoryMaxValues.isEmpty())
+            return null;
+
+        double coreAverage = 0.0;
+        if (!acceleratorCoreMaxValues.isEmpty())
+            coreAverage = CommonUtils.percentile(PERFORMANCE_ACCELERATOR_PERCENTILE, acceleratorCoreMaxValues);
+
+        double memoryAverage = 0.0;
+        if (!acceleratorMemoryMaxValues.isEmpty())
+            memoryAverage = CommonUtils.percentile(PERFORMANCE_ACCELERATOR_PERCENTILE, acceleratorMemoryMaxValues);
```

Comment:
Same here, remove the redundant checks

Reply:
Thanks for the review @dinogun . In the earlier check we `continue` if both of the lists are empty. at the time of calculating percentile we check if the respective list is not empty. So I feel we will be needing this check.